### PR TITLE
reduce customData overhead via streamlined boilerplate

### LINKED ISSUE #4149: Deployment failed  due to invalid customdata
**Is this a request for help?**: YES

---

**Is this an ISSUE or FEATURE REQUEST?** (choose one): Issue

---

**What version of acs-engine?**: latest:
```
Version: v0.24.3
GitCommit: ad007b644
GitTreeState: clean
```

---

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
Kubernetes 1.11

**What happened**:
Tried to deploy new cluster. Deployment failed with the following error:
```
FATA[0272] Code="DeploymentFailed" Message="At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-debug for usage details." Details=[{"code":"BadRequest","message":"{\r\n  \"error\": {\r\n    \"code\": \"InvalidParameter\",\r\n    \"message\": \"Custom data in OSProfile must be in Base64 encoding and with a maximum length of 87380 characters.\",\r\n    \"target\": \"customData\"\r\n  }\r\n}"}]
```

**What you expected to happen**:
Successful deployment

**How to reproduce it** (as minimally and precisely as possible):
This is my apimodel.json I used for the deployment:
```
{
  "apiVersion": "vlabs",
  "properties": {
    "orchestratorProfile": {
      "orchestratorType": "Kubernetes",
      "orchestratorRelease": "1.11",
      "kubernetesConfig": {
        "useManagedIdentity": false,
        "enableDataEncryptionAtRest": true,
        "dockerEngineVersion": "17.03.*",
        "apiServerConfig": {
          "--admission-control":  "NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultStorageClass,DefaultTolerationSeconds,MutatingAdmissionWebhook,ValidatingAdmissionWebhook,ResourceQuota,DenyEscalatingExec,AlwaysPullImages,ValidatingAdmissionWebhook,ResourceQuota"
        },
        "kubeletConfig": {
          "--max-pods": "100"
        },
        "addons": [
          {
            "name": "cluster-autoscaler",
            "enabled": true,
            "config": {
              "minNodes": "3",
              "maxNodes": "10"
            }
          },
          {
            "name": "aad-pod-identity",
            "enabled": true
          },
          {
            "name": "keyvault-flexvolume",
            "enabled" : true
          },
          {
            "name": "tiller",
            "enabled" : false
          },
          {
              "name": "kubernetes-dashboard",
              "enabled" : false
          },
          {
            "name": "container-monitoring",
            "enabled": true,
            "containers": [
              {
                "name": "omsagent",
                "image": "microsoft/oms:acsenginelogfixnew",
                "cpuRequests": "50m",
                "memoryRequests": "200Mi",
                "cpuLimits": "150m",
                "memoryLimits": "750Mi"
              }
            ],
            "config": {
              "dockerProviderVersion": "2.0.0-3",
              "omsAgentVersion": "1.6.0-42"
            }
          }
        ]
      }
    },
    "masterProfile": {
      "count": 5,
      "vmSize": "Standard_D2_v2",
      "vnetSubnetId": "<>",
      "agentVnetSubnetID": "<>",
      "firstConsecutiveStaticIP": "<>,
      "vnetCidr": "<>",
      "availabilityProfile": "VirtualMachineScaleSets"
    },
    "agentPoolProfiles": [
      {
        "name": "devops",
        "count": 2,
        "vmSize": "Standard_F8s_v2",
        "availabilityProfile": "VirtualMachineScaleSets",
        "storageProfile": "ManagedDisks",
        "vnetSubnetId": "<>",
        "ipAddressCount": 100
      }
    ],
    "linuxProfile": {
      "adminUsername": "<>",
      "ssh": {
        "publicKeys": [
          {
	          "keyData": ""
          }
        ]
      }
    },
    "servicePrincipalProfile": {
      "clientId": "",
      "secret": "",
      "objectId": ""
    },
    "aadProfile": {
      "serverAppID": "<>",
      "clientAppID": "<>",
      "tenantID": "<>",
      "adminGroupID": "<>"
    }
  }
}

```

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by jackfrancis**
**What this PR does / why we need it**:

- remove unnecessary `"` wrapping around yaml strings
- remove unnecessary `"` chars in shell scripts
- reduce chars in to-be-sed'd sentinel tokens
- optimize multiple sed statements against common files

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #4149

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```
reduce customData overhead via streamlined boilerplate
```

### CODE REVIEW

**Comment by jackfrancis** [comment]
Comparing control plane `customData` of a sample test in this PR (the k8s-1.10-release-e2e above) to a test before this change for the same test, I see a 4k byte savings.

File: parts/k8s/kubernetesmastercustomdata.yml [hunk 1]
```diff
@@ -280,174 +280,118 @@ MASTER_ARTIFACTS_CONFIG_PLACEHOLDER
...
 {{end}}
 {{if eq .OrchestratorProfile.KubernetesConfig.NetworkPlugin "flannel"}}
-    sed -i "s|<kubeClusterCidr>|{{WrapAsParameter "kubeClusterCidr"}}|g" "/etc/kubernetes/addons/flannel-daemonset.yaml"
+    sed -i "s|<kubeClusterCidr>|{{WrapAsParameter "kubeClusterCidr"}}|g" /etc/kubernetes/addons/flannel-daemonset.yaml
 {{end}}
 {{if eq .OrchestratorProfile.KubernetesConfig.NetworkPolicy "cilium"}}
+  a=/etc/kubernetes/addons/cilium-daemonset.yaml
   {{if IsMasterVirtualMachineScaleSets}}
-    sed -i "s|<ETCD_URL>|"https://$PRIVATE_IP:$ETCD_CLIENT_PORT"|g" "/etc/kubernetes/addons/cilium-daemonset.yaml"
+    sed -i "s|<ETCD_URL>|"https://$PRIVATE_IP:$ETCD_CLIENT_PORT"|g" $a
```
Comment (CecileRobertMichon) [inline parts/k8s/kubernetesmastercustomdata.yml]: what's `$a`?

Comment (jackfrancis) [reply parts/k8s/kubernetesmastercustomdata.yml]: **a**... really good way to save space?

actually a=*a*ddon filepath

Comment (CecileRobertMichon) [reply parts/k8s/kubernetesmastercustomdata.yml]: yeah my bad I just saw it was declared above, I was wondering where it was defined.

reading is hard

**Pull request merged**